### PR TITLE
Staging Changes

### PR DESCRIPTION
**Pull request by Rayahhhmed**
* Cleaned up auth redirects to be dependent on Environment.
* Cleaned up auth comments on the referencing bugs.
* Merging in @hhuolu 's Logout dialogue.
* Auth users added to Database and db integrity finalised for addition of new users.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (lhjt) [inline .github/workflows/ci.yml:1]: I am assuming the changes to this file are intentional and u know what ur doing

**Review by lhjt — APPROVED** [review]

File: server/src/auth/auth.service.ts [hunk 1]
```diff
@@ -13,7 +14,8 @@ export class AuthService {
     );
 
     if (!id_token || !TrustIssuer) {
-      return res.redirect(REDIRECT_LINK);
+      console.log(this.configService.get<string>('app.redirectLink'));
```
Comment (Shzmj) [inline server/src/auth/auth.service.ts]: debug print statement still here

File: server/src/auth/auth.controller.ts [hunk 2]
```diff
@@ -1,21 +1,33 @@
...
 
   @Get('/user')
-  user(@Request() req, @Res() res: Response) {
+  async user(@Request() req, @Res() res: Response) {
     if (req.user) {
+      const userID = req.user.userinfo.sub;
+      try {
+        await this.userService.getUserInfo(userID);
+      } catch (e) {
+        console.debug(`User ${userID} does not exist in db, adding them now!`);
```
Comment (Shzmj) [inline server/src/auth/auth.controller.ts:28]: remove debugging statements

File: client/src/components/sidebar/UserAccount.tsx [hunk 3]
```diff
@@ -112,29 +112,64 @@ const UserAccount: React.FC<UserAccountProps> = ({ collapsed }) => {
...
-    </UserAuth>
+        ) : (
+          <>
+            <UserInfo>
+              <StyledAccountIcon />
+              {/* TODO: handle user's name */}
+              <p>{user.zid}</p>
+            </UserInfo>
+            <Tooltip title="Log out" placement="right">
+              {/* TODO: error handling for when logging out */}
```
Comment (Shzmj) [inline client/src/components/sidebar/UserAccount.tsx]: are the todos intentional i.e. will it be implemented in the future?

**Review by Shzmj — CHANGES_REQUESTED** [review]

File: server/src/auth/auth.service.ts (hunk 1 above)
Comment (Rayahhhmed) [reply server/src/auth/auth.service.ts]: o this is legit error

File: server/src/auth/auth.controller.ts (hunk 2 above)
Comment (Rayahhhmed) [reply server/src/auth/auth.controller.ts:28]: Nah need this for dev logs debugging

File: client/src/components/sidebar/UserAccount.tsx (hunk 3 above)
Comment (Rayahhhmed) [reply client/src/components/sidebar/UserAccount.tsx]: Todos necessary for future

**Review by Shzmj — APPROVED** [review]

**Pull request merged**